crates/ruff_linter/src/rules/pyupgrade/rules/use_pep695_type_alias.rs

```diff
@@ -92,20 +94,27 @@ pub(crate) fn non_pep695_type_alias(checker: &mut Checker, stmt: &StmtAnnAssign)
 
     // TODO(zanie): We should check for generic type variables used in the value and define them
     //              as type params instead
-    let mut diagnostic = Diagnostic::new(NonPEP695TypeAlias { name: name.clone() }, stmt.range());
-    let mut visitor = TypeVarReferenceVisitor {
-        vars: vec![],
-        semantic: checker.semantic(),
+    let vars = {
+        let mut visitor = TypeVarReferenceVisitor {
+            vars: vec![],
+            semantic: checker.semantic(),
+        };
+        visitor.visit_expr(value);
+        visitor.vars
     };
-    visitor.visit_expr(value);
 
-    let type_params = if visitor.vars.is_empty() {
+    // Type variables must be unique; filter while preserving order.
+    let vars = vars
+        .into_iter()
+        .unique_by(|TypeVar { name, .. }| name.id.as_str())
+        .collect::<Vec<_>>();
```

Comment:
Since `Itertools` is already in scope, you could avoid the turbofish here:

```suggestion
        .collect_vec();
```

Reply:
As a habit, I try to avoid using `Itertools` where it's not providing much value since it (1) makes it a little easier if we remove the dep in the future and (2) in my opinion at least it's actually _easier_ to read the explicit version.